### PR TITLE
[Bug] Fix stream load UT failed

### PR DESCRIPTION
**Pull request by weizuo93**
## Proposed changes

Fix stream load UT failed.
```
==9026==ERROR: AddressSanitizer: SEGV on unknown address 0x0000000003d8 (pc 0x000001f4f7ec bp 0x7ffe39bda4f0 sp 0x7ffe39bda4e0 T0)
==9026==The signal is caused by a READ memory access.
==9026==Hint: address points to the zero page.
    #0 0x1f4f7ec in std::__shared_ptr<doris::StreamLoadRecorder, (__gnu_cxx::_Lock_policy)2>::__shared_ptr(std::__shared_ptr<doris::StreamLoadRecorder, (__gnu_cxx::_Lock_policy)2> const&) (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x1f4f7ec)
    #1 0x1f4f89b in std::shared_ptr<doris::StreamLoadRecorder>::shared_ptr(std::shared_ptr<doris::StreamLoadRecorder> const&) (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x1f4f89b)
    #2 0x1f4f908 in doris::StorageEngine::get_stream_load_recorder() (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x1f4f908)
    #3 0x1f3c5d6 in doris::StreamLoadAction::_sava_stream_load_record(doris::StreamLoadContext*, std::string const&) /root/doris/incubator-doris/be/src/http/action/stream_load.cpp:546
    #4 0x1f33b22 in doris::StreamLoadAction::on_header(doris::HttpRequest*) /root/doris/incubator-doris/be/src/http/action/stream_load.cpp:239
    #5 0x160ab35 in doris::StreamLoadActionTest_no_auth_Test::TestBody() /root/doris/incubator-doris/be/test/http/stream_load_test.cpp:109
    #6 0x4681759 in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x4681759)
    #7 0x467c709 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x467c709)
    #8 0x4662db7 in testing::Test::Run() (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x4662db7)
    #9 0x46635e1 in testing::TestInfo::Run() (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x46635e1)
    #10 0x4663c2e in testing::TestCase::Run() (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x4663c2e)
    #11 0x466a499 in testing::internal::UnitTestImpl::RunAllTests() (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x466a499)
    #12 0x4682736 in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x4682736)
    #13 0x467d3db in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x467d3db)
    #14 0x46691df in testing::UnitTest::Run() (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x46691df)
    #15 0x161a384 in RUN_ALL_TESTS() (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x161a384)
    #16 0x160f193 in main /root/doris/incubator-doris/be/test/http/stream_load_test.cpp:271
    #17 0x7f86b95fc504 in __libc_start_main (/lib64/libc.so.6+0x22504)
    #18 0x153c44d  (/root/doris/incubator-doris/be/ut_build_ASAN/test/http/stream_load_test+0x153c44d)

```

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

### CODE REVIEW

**Review by morningman — DISMISSED** [review]
LGTM

File: be/src/olap/storage_engine.cpp [hunk 1]
```diff
@@ -227,15 +227,19 @@ Status StorageEngine::_init_store_map() {
...
+
+    RETURN_NOT_OK_STATUS_WITH_WARN(_init_stream_load_recorder(stream_load_record_path),
+                                   "init StreamLoadRecorder failed");
 
     return Status::OK();
 }
 
-Status StorageEngine::_init_stream_load_recorder() {
-    std::string stream_load_record_path = config::stream_load_record_path;
+Status StorageEngine::_init_stream_load_recorder(std::string stream_load_record_path) {
```
Comment (morningman) [inline be/src/olap/storage_engine.cpp:241]:
```suggestion
Status StorageEngine::_init_stream_load_recorder(const std::string& stream_load_record_path) {
```

Comment (weizuo93) [reply be/src/olap/storage_engine.cpp]: > 

OK.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**